draft-ietf-quic-transport.md

```diff
@@ -336,175 +336,327 @@ all field sizes are in bits.  When discussing individual bits of fields, the
+Different packet types explicitly indicate the encryption level for ease of
+decryption. These packets contain:
+* Octet 0: 0x85, 0x86 or 0x87
+* Octets 1-8: Connection ID (initial or final)
+* Octets 9-12: Packet Number
+* Octets 13-16: Version
+* Octets 17+: Encrypted Payload
+
+A first octet of 0x85 indicates a 0-RTT packet. After the 1-RTT keys are
+established, key phases are used by the QUIC packet protection to identify the
+correct packet protection keys. The default initial value key phase is 0. See
+{{QUIC-TLS}} for more details.
+
+The encrypted payload is both authenticated and encrypted using packet
+protection keys. {{QUIC-TLS}} describes packet protection in detail.  After
+decryption, the plaintext consists of a sequence of frames, as described in
+{{frames}}.
+
+
+## Public Reset Packet {#public-reset-packet}
+
+A Public Reset packet is sent by only a server and is used to abruptly terminate
+communications. Public Reset is provided as an option of last resort for a
+server that does not have access to the state of a connection.  This is intended
+for use by a server that has lost state (for example, through a crash or
+outage). A server that wishes to communicate a fatal connection error MUST use a
+CONNECTION_CLOSE frame if it has sufficient state to do so.
+
+A Public Reset packet contains:
+
+* Octet 0: 0x88
+* Octets 1-8: Echoed data (octets 1-8 of received packet)
+* Octets 9-12: Echoed data (octets 9-12 of received packet)
+* Octets 13-16: Version
+* Octets 17+: Public Reset Proof
+
+For a client that sends a connection ID on every packet, the Connection ID field
+is simply an echo of the initial Connection ID, and the Packet Number field
+includes an echo of the client's packet number (and, depending on the client's
+packet number length, 0, 2, or 3 additional octets from the client's packet).
+
+A Public Reset packet sent by a server indicates that it does not have the
+state necessary to continue with a connection.  In this case, the server will
+include the fields that prove that it originally participated in the connection
+(see {{public-reset-proof}} for details).
+
+Upon receipt of a Public Reset packet that contains a valid proof, a client
+MUST tear down state associated with the connection.  The client MUST then
+cease sending packets on the connection and SHOULD discard any subsequent
+packets that arrive. A Public Reset that does not contain a valid proof MUST be
+ignored.
+
+### Public Reset Proof
+
+TODO: Details to be added.
+
+
+## Connection ID {#connection-id}
+
+QUIC connections are identified by their 64-bit Connection ID. All long headers
+contain a Connection ID. Short headers indicate the presence of a Connection ID
+using the CONNECTION_ID flag. When present, the Connection ID is in the same
+location in all packet headers, making it straightforward for middleboxes, such
+as load balancers, to locate and use it.
 
+When a connection is initiated, the client MUST choose a random value and use it
```

Comment:
The client then provides a suggestion that may be reasonably for servers to simply use. Only a server that cares about structure in the ID changes it, others use the clients suggested value as is.